### PR TITLE
[BackPort from Master] Add Vhosts per organization in Choreo

### PR DESCRIPTION
**Pull request by RAVEENSR**
This PR back-ports the changes from the following PRs to 9.2.x branch:

https://github.com/wso2/carbon-apimgt/pull/10855
https://github.com/wso2/carbon-apimgt/pull/11051
https://github.com/wso2/carbon-apimgt/pull/11116

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/EnvironmentsApiServiceImpl.java [hunk 1]
```diff
@@ -95,9 +96,10 @@ public Response environmentsGet(MessageContext messageContext) throws APIManagem
     public Response environmentsPost(EnvironmentDTO body, MessageContext messageContext) throws APIManagementException {
         try {
             APIAdmin apiAdmin = new APIAdminImpl();
-            String tenantDomain = RestApiCommonUtil.getLoggedInUserTenantDomain();
+            //String tenantDomain = RestApiCommonUtil.getLoggedInUserTenantDomain();
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/EnvironmentsApiServiceImpl.java:99]: remove

File: components/apimgt/org.wso2.carbon.apimgt.internal.service/src/main/java/org/wso2/carbon/apimgt/internal/service/impl/ApisApiServiceImpl.java [hunk 2]
```diff
@@ -94,6 +95,7 @@ public Response deployedAPIRevision(List<DeployedAPIRevisionDTO> deployedAPIRevi
         APIProvider apiProvider = RestApiCommonUtil.getLoggedInUserProvider();
 
         List<String> revisionUUIDs = new ArrayList<>();
+        String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.internal.service/src/main/java/org/wso2/carbon/apimgt/internal/service/impl/ApisApiServiceImpl.java:98]: we can use APIInfo getAPIInfoByUUID(String id) throws APIManagementException  in APIProvider from that get the organization id

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/EnvironmentsApiServiceImpl.java [hunk 3]
```diff
@@ -40,8 +40,9 @@ public class EnvironmentsApiServiceImpl implements EnvironmentsApiService {
      */
     public Response environmentsEnvironmentIdDelete(String environmentId, MessageContext messageContext) throws APIManagementException {
         APIAdmin apiAdmin = new APIAdminImpl();
-        String tenantDomain = RestApiCommonUtil.getLoggedInUserTenantDomain();
-        apiAdmin.deleteEnvironment(tenantDomain, environmentId);
+        //String tenantDomain = RestApiCommonUtil.getLoggedInUserTenantDomain();
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/EnvironmentsApiServiceImpl.java:43]: lets remove comment

**Review by malinthaprasan — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/EnvironmentsApiServiceImpl.java (hunk 1 above)
Comment (RAVEENSR) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/EnvironmentsApiServiceImpl.java]: Fixed with [9359817](https://github.com/wso2/carbon-apimgt/pull/11227/commits/9359817b8246e036175544e876bdffcaf54e6d71)

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/EnvironmentsApiServiceImpl.java (hunk 3 above)
Comment (RAVEENSR) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/EnvironmentsApiServiceImpl.java]: Fixed with [9359817](https://github.com/wso2/carbon-apimgt/pull/11227/commits/9359817b8246e036175544e876bdffcaf54e6d71)

**Pull request merged**